clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -238,7 +238,7 @@ public RecordsInfo info() {
         }
     }
 
-    public void setProducerState(long producerId, short producerEpoch, int baseSequence) {
+    public void setProducerState(long producerId, short producerEpoch, int baseSequence, boolean isTransactional) {
```

Comment:
I think this is a good general improvement, to set all this data right at the very end.